File: Data Structures & Algorithms/two-integer-sum/submission-1.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> pairs = new Dictionary<int, int>();
        int[] result = {0, 0};

        for (int i = 0; i < nums.Length; i++)
        {
            if (pairs.ContainsKey(-nums[i]))
            {
                result = new int[] {pairs[-nums[i]], i};
                break;
            }
            else
            {
                pairs.Add(nums[i] - target, i);
            }
        }
        return result;
    }
}

File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = -1
        r = len(nums)
        m = l + ((r - l) // 2)
        while l < r-1:
            if nums[m] < target:
                l = m
            elif nums[m] > target:
                r = m
            else:
                return m
            m = l + ((r - l) // 2)
        if nums[m] != target:
            return -1
        return m